bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -67,6 +67,9 @@ def run_plan(scope, plan_name, named_args = {})
     # The perspective of the environment is wanted here (for now) to not have to
     # require modules to have dependencies defined in meta data.
     loader = loaders.private_environment_loader
+    # Load resource types (full path commented below, TODO: delete)
+    # closure_scope.compiler.loaders.static_loader.runtime_3_init
+    loaders.static_loader.runtime_3_init
```

Comment:
Unsure if this will load resource types on the module path as well. Still need to test.

Reply:
Does this belong in our PAL class?

Reply:
I would expect this to probably go next to `alias_types`. https://github.com/puppetlabs/bolt/blob/master/lib/bolt/pal.rb#L109

Reply:
I ended up doing it here because I did not see a way around using some hackery next to `alias_types` (something like: `compiler.instance_variable_get(:@internal_compiler).loaders.static_loader.runtime_3_init`)

HOWEVER Could the solution for this to be better handled in puppet? Specifically by **not** treating `Puppet[:tasks]]` special? 
https://github.com/puppetlabs/puppet/blob/904023ffc59bc6771c0c7abf2a1d2c4acf941b09/lib/puppet/pops/loaders.rb#L383-L397

Reply:
Update, we do want it in PAL class. We do not want to do it in puppet because of the risk of using the 3x loaders.